### PR TITLE
Repository sync role

### PR DESCRIPTION
**Pull request by inderpaltiwana**
### What does this PR do?
Creating a repository_sync role

### How should this be tested?
Automated tests are preferred, but not always doable - especially for infrastructure. Include commands to run your new feature, and also post-run commands to validate that it worked. (please use code blocks to format code samples)

### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.
resolves #<number>

### Other Relevant info, PRs, etc.
Please provide link to other PRs that may be related (blocking, resolves, etc. etc.)

### CODE REVIEW

File: roles/repository_sync/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+
+# Sync AH repository
+- name: Sync Automation Hub repository
+  ah_repository_sync:
+    name:                       "{{ __repository_item.name }}"
+    wait:                        "{{ __repository_item.wait | default(omit)  }}"
+    auth_url:                   "{{ __repository_item.interval | default(1) }}"
```
Comment (djdanielsson) [inline roles/repository_sync/tasks/main.yml]: is this correct? auth_url doesn't seem correct here.

**Review by djdanielsson — CHANGES_REQUESTED** [review]
I have a question about one line

File: roles/repository_sync/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    timeout:                      "{{ __repository_item.timeout | default(omit) }}"
+  loop: "{{ ah_repositories }}"
+  loop_control:
+    loop_var: "__repository_item"
+  no_log: "{{ ah_configuration_repository_secure_logging }}"
```
Comment (djdanielsson) [inline roles/repository_sync/tasks/main.yml:13]: after a closer look, I am not sure we want this var for no_log I think you have it correct in the async job ah_configuration_repository_syncs_secure_logging

File: roles/repository_sync/tasks/main.yml (hunk 1 above)
Comment (Tompage1994) [reply roles/repository_sync/tasks/main.yml]: Should be `interval`

File: roles/repository_sync/tasks/main.yml (hunk 2 above)
Comment (Tompage1994) [reply roles/repository_sync/tasks/main.yml:13]: This is correct in the same way we have done for the other roles (e.g. `ee_repository_sync`). The idea here was that you would use the same var for the related roles.

**Review by Tompage1994 — CHANGES_REQUESTED** [review]

File: roles/repository_sync/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    jid: "{{ __repositories_syncs_job_async_result_item.ansible_job_id }}"
+  register: __repositories_syncs_job_async_result
+  until: __repositories_syncs_job_async_result.finished
+  retries: "{{ ah_configuration_repository_syncs_async_retries }}"
+  delay: "{{ ah_configuration_repository_syncs_async_delay }}"
+  loop: "{{ __repositories_syncs_job_async.results }}"
+  loop_control:
+    loop_var: __repositories_syncs_job_async_result_item
+  when: __repositories_syncs_job_async_result_item.ansible_job_id is defined
+  no_log: "{{ ah_configuration_repository_syncs_secure_logging }}"
```
Comment (Tompage1994) [inline roles/repository_sync/tasks/main.yml]: See what you mean. This one needs to change to `ah_configuration_repository_secure_logging`

```suggestion
  no_log: "{{ ah_configuration_repository_secure_logging }}"
```

File: roles/repository_sync/tasks/main.yml (hunk 2 above)
Comment (Tompage1994) [reply roles/repository_sync/tasks/main.yml:13]: I see what you mean. Its the other line that's wrong. I've added a suggested line change

File: roles/repository_sync/tasks/main.yml (hunk 3 above)
Comment (djdanielsson) [reply roles/repository_sync/tasks/main.yml]: No, ah_configuration_repository_secure_logging is already used in a role which is why I was trying to say I think it should be ah_configuration_repository_syncs_secure_logging instead

Comment (Tompage1994) [reply roles/repository_sync/tasks/main.yml]: Exactly, it should be the same var for the `repository` and `repository_sync` roles

Comment (Tompage1994) [reply roles/repository_sync/tasks/main.yml]: See that ee_registry_X uses the same idea:

https://github.com/redhat-cop/ah_configuration/blob/e1dfbab02ef95afd54f2d03c7c4bcb60988129f9/roles/ee_registry/tasks/main.yml#L26

https://github.com/redhat-cop/ah_configuration/blob/e1dfbab02ef95afd54f2d03c7c4bcb60988129f9/roles/ee_registry_index/tasks/main.yml#L18

File: roles/repository_sync/tasks/main.yml (hunk 1 above)
Comment (inderpaltiwana) [reply roles/repository_sync/tasks/main.yml]: Yep didn't spot this before pushing up, will make relevant changes

File: roles/repository_sync/tasks/main.yml (hunk 3 above)
Comment (inderpaltiwana) [reply roles/repository_sync/tasks/main.yml]: Updated with `ah_configuration_repository_secure_logging` as suggested

**Comment by inderpaltiwana** [comment]
Have updated this PR based on the comments and suggestions from reviewers

**Review by Tompage1994 — APPROVED** [review]

**Comment by Tompage1994** [comment]
LGTM. @djdanielsson Let me know if you disagree

**Review by djdanielsson — APPROVED** [review]
LGTM

**Pull request merged**